File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

map<string,int> m;
int pos;
vector<int> v[1001];
bool visit[1001][1001];

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> answer;
    for(int i=0;i<id_list.size();i++)
    {
        m[id_list[i]] = i;
        answer.push_back(0);
    }
    for(int i=0;i<report.size();i++)
    {
        string s= report[i];
        pos = report[i].find(" ",0);
        string a= report[i].substr(0,pos);
        string b = report[i].substr(pos+1,report[i].size());
        if(visit[m[a]][m[b]])
            continue;
        v[m[b]].push_back(m[a]);
        visit[m[a]][m[b]] = 1;
    }
    for(int i=0;i<id_list.size();i++)
    {
        if(v[i].size()<k)
            continue;
        for(auto a : v[i])
        {
            answer[a]++;
        }
    }
    return answer;
}

File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.swift
import Foundation

func solution(_ id_list:[String], _ report:[String], _ k:Int) -> [Int] {
    var s = Set(report)
    var dic = [String: [String]]()
    var accumulateDic = [String: Int]()
    var result = [Int]()

    for i in s {
        let temp = i.split(separator: " ").map { String($0) }
        if accumulateDic[temp[1]] == nil {
            accumulateDic[temp[1]] = 1
        } else {
            accumulateDic[temp[1]]! += 1
        }
        if dic[temp[0]] == nil {
            dic[temp[0]] = [temp[1]]
        } else {
            dic[temp[0]]?.append(temp[1])
        }
    }

    id_list.forEach {
        var cnt = 0
        if let dic = dic[$0] {
            for i in dic {
                if accumulateDic[i]! >= k {
                    cnt += 1
                }
            }
        }
        result.append(cnt)
    }
    return result
}